Source/Blazorise/ComponentDisposer.cs

```diff
@@ -65,10 +65,11 @@ public ComponentDisposer( IServiceProvider serviceProvider )
         private IList<object> LoadServiceProviderDisposableList()
         {
             var disposablesPropertyInfo = ServiceProvider.GetType().GetProperty( "Disposables", BindingFlags.Instance | BindingFlags.NonPublic );
+            var disposables = disposablesPropertyInfo?.GetValue( ServiceProvider ) as IList<object> ?? Array.Empty<object>();
 
-            disposePossible = disposablesPropertyInfo is not null;
+            disposePossible = disposables is not null && !disposables.GetType().IsArray;
```

Comment:
Why is this `IsArray` check? Seems like an hack, what is it for? Checking for Empty? Would `!disposables.IsNullOrEmpty()` not be the same and more clear?

Reply:
Because internally, disposables could be an array https://github.com/dotnet/runtime/blob/cb52e3b88177cb86929b829d92b03561ad4e806b/src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs#L15

Reply:
It's still a (`IList<object>`) in the end. `IsNullOrEmpty` should work always.